src/raw.rs

```diff
@@ -584,3 +621,49 @@ pub fn get_keyspace_events() -> NotifyEvent {
         NotifyEvent::from_bits_truncate(events)
     }
 }
+
+#[allow(clippy::not_unsafe_ptr_arg_deref)]
+pub fn is_io_error(rdb: *mut RedisModuleIO) -> bool {
+    unsafe { RedisModule_IsIOError.unwrap()(rdb) != 0 }
+}
+
+pub struct Version {
+    pub major: c_int,
+    pub minor: c_int,
+    pub patch: c_int,
+}
+
+impl From<c_int> for Version {
+    fn from(ver: c_int) -> Self {
+        // Expected format: 0x00MMmmpp for Major, minor, patch
+        Version {
+            major: (ver & 0x00FF_0000) >> 16,
+            minor: (ver & 0x0000_FF00) >> 8,
+            patch: ver & 0x0000_00FF,
+        }
+    }
+}
+
+#[allow(clippy::not_unsafe_ptr_arg_deref)]
+pub fn get_redis_version() -> Version {
+    unsafe { Version::from(RedisModule_GetServerVersion.unwrap()()) }
+}
+
+pub fn check_minimal_version_for_short_read() -> bool {
```

Comment:
@oshadmi modules without aux data can use short read even before this version, I suggest leaving it for the module to check ...